modules/library/src/main/scala/zio/elasticsearch/HttpElasticExecutor.scala

```diff
@@ -259,33 +315,87 @@ private[elasticsearch] final class HttpElasticExecutor private (config: ElasticC
+    val requestBody     = r.query.toJson merge createPointInTimeJson(pitId) merge defaultSortField
     sendRequestWithCustomResponse(
       request
-        .post(uri"${config.uri}/$Search/$Scroll".withParams((Scroll, ScrollDefaultDuration)))
+        .get(uri"${config.uri}/$Search")
         .response(asJson[ElasticQueryResponse])
         .contentType(ApplicationJson)
-        .body(Obj(ScrollId -> Str(scrollId)))
+        .body(searchAfterJson.map(_ merge requestBody).getOrElse(requestBody))
     ).flatMap { response =>
       response.code match {
         case HttpOk =>
           response.body.fold(
             e => ZIO.fail(new ElasticException(s"Exception occurred: ${e.getMessage}")),
-            value =>
-              if (value.results.isEmpty) ZIO.succeed((Chunk.empty, None))
-              else ZIO.succeed((Chunk.fromIterable(value.results).map(Item), value.scrollId.orElse(Some(scrollId))))
+            body => {
+              if (body.results.isEmpty)
+                ZIO.succeed((Chunk.empty, None))
+              else
+                body.pitId match {
+                  case Some(newPitId) =>
+                    body.lastSortField match {
+                      case Some(newSearchAfter) =>
+                        ZIO.succeed((Chunk.fromIterable(body.results.map(Item)), Some(newPitId, Some(newSearchAfter))))
+                      case None =>
+                        ZIO.fail(
+                          new ElasticException(
+                            s"Unexpected response from Elasticsearch, search_after field is missing."
+                          )
+                        )
+                    }
+                  case None =>
+                    ZIO.fail(
+                      new ElasticException(
+                        s"Unexpected response from Elasticsearch, pid.id field is missing."
+                      )
+                    )
+                }
+            }
           )
         case _ =>
           ZIO.fail(createElasticExceptionFromCustomResponse(response))
       }
     }
+  }
+
+  private def createPointInTimeJson(pitId: String, keepAliveDuration: Option[String] = None) =
+    Json.Obj(
+      "pit" -> Json.Obj(
+        "id"      -> Json.Str(pitId),
+        KeepAlive -> Json.Str(keepAliveDuration.getOrElse(KeepAliveDefaultDuration))
+      )
+    )
+
+  private def createElasticException(response: Response[Either[String, String]]): ElasticException =
+    new ElasticException(
+      s"Unexpected response from Elasticsearch. Response body: ${response.body.fold(body => body, _ => "")}"
+    )
+
+  private def createElasticExceptionFromCustomResponse[A](
+    response: Response[Either[ResponseException[String, String], A]]
+  ): ElasticException =
+    new ElasticException(
+      s"Unexpected response from Elasticsearch. Response body: ${response.body.fold(body => body, _ => "")}"
+    )
+
+  private lazy val defaultSortField =
+    Json.Obj("sort" -> Json.Arr(Json.Str(Doc)))
```

Comment:
Are you using `_doc` or `_id` here? Also, this can be moved down as a constant.

Reply:
I added changes and then reverted them. For optimisation it is best to have "_shard_doc" as sort value, since we do not care about sorting at this time we should go with optimisation.